src/currentprocess.rs

```diff
@@ -195,17 +199,45 @@ pub trait ProcessSource {
     /// Real process ids are <= u32::MAX.
     /// Test process ids are > u32::MAX
     fn id(&self) -> u64;
+    fn is_stderr_a_tty(&self) -> bool;
+    fn is_stdout_a_tty(&self) -> bool;
 }
 
 // ----------- real process -----------------
 
-#[derive(Clone, Debug, Default)]
-pub struct OSProcess {}
+#[derive(Clone, Debug)]
+pub struct OSProcess {
+    is_stderr_a_tty: bool,
```

Comment:
These clearly have to be here to be cached, but I think we can make the accessing functions live on Writer.

Then e.g.

```
impl Writer for io::Stdout {
    ...
    fn is_a_tty(&self) -> bool {
        process().is_stdout_a_tty()
    }
```

... except that that won't work...
... except that it *will* if we merge PR 3368 first.

This can be done as an evolution on top of your PR of course, no need to hold it up.